modules/ztp-pre-staging-downloading-artifacts.adoc

```diff
@@ -0,0 +1,367 @@
+multicluster-engine                                openshift-pipelines-operator-rh.v1.6.4       Red Hat OpenShift Pipelines                  1.6.4                 openshift-pipelines-operator-rh.v1.6.3            Succeeded
+----
+
+. Query the AI version on your hub cluster:
++
+[source,terminal]
+----
+$ oc get cm assisted-service -n open-cluster-management -oyaml | \ <1>
+grep -E "AGENT_DOCKER_IMAGE|CONTROLLER_IMAGE|INSTALLER_IMAGE"
+----
+<1> The namespace can differ based on the installed {rh-rhacm} version.
+
+. Save the output, which is required for downloading the artifacts.
+
++
+.Example output for AI version
+[source,terminal]
+----
+  AGENT_DOCKER_IMAGE: registry.redhat.io/multicluster-engine/assisted-installer-agent-rhel8@sha256:da1753f9fcb9e229d0a68de03fac90d15023e647a8db531ae489eb93845d5306
+  CONTROLLER_IMAGE: registry.redhat.io/multicluster-engine/assisted-installer-reporter-rhel8@sha256:e8d6b78248352b1a8e05a22308185a468d4a139682d997a7f968b329abbc02cd
+  INSTALLER_IMAGE: registry.redhat.io/multicluster-engine/assisted-installer-rhel8@sha256:33abd6e21cfdc36dd4337fa6f3c3442d33fc3f976471614dca5b8ef749e7a027
+----
+
+. To access the container registry, copy a valid pull secret on the server to be installed:
+
+.. Create the `.docker` folder:
++
+[source,terminal]
+----
+$ mkdir /root/.docker
+----
+
+.. Copy the `config.json` file to the previously created `.docker` folder:
++
+[source,terminal]
+----
+$ cp config.json /root/.docker/config.json <1>
+----
+<1> `/root/.docker/config.json` is the default path where `podman` checks for the login credentials for the registry.
+
+[NOTE]
+====
+If you use a different registry to pull the required artifacts, you need to copy the proper pull secret.
+If the local registry uses TLS, you need to include the certificates from the registry as well.
+====
+
+[id="ztp-downloading-ocp-images_{context}"]
+== Downloading the {product-title} images
+
+The {factory-prestaging-tool} allows you to precache all the container images required to provision a specific {product-title} release.
+
+.Procedure
+
+. Pre-cache the release by running the following command:
++
+[source,terminal]
+----
+# podman run -v /mnt:/mnt -v /root/.docker:/root/.docker --privileged --rm quay.io/openshift-kni/telco-ran-tools -- \
+   factory-precaching-cli download -r 4.11.5 -f /mnt \
+   --ai-img registry.redhat.io/multicluster-engine/assisted-installer-agent-rhel8@sha256:da1753f9fcb9e229d0a68de03fac90d15023e647a8db531ae489eb93845d5306 \
+   --ai-img registry.redhat.io/multicluster-engine/assisted-installer-reporter-rhel8@sha256:e8d6b78248352b1a8e05a22308185a468d4a139682d997a7f968b329abbc02cd \
+   --ai-img registry.redhat.io/multicluster-engine/assisted-installer-rhel8@sha256:33abd6e21cfdc36dd4337fa6f3c3442d33fc3f976471614dca5b8ef749e7a027 \
+   --img quay.io/user/troubleshoot
```

Comment:
@alosadagrande Based on your other comments, I assume any `quay.io/user/troubleshoot` instances are incorrect as well? What does the factory-cli `--img` flag do? Is this where the images can be found?

Reply:
no, that's ok. The --img flag basically pre-caches any image you set there. The idea is for instance if you have some custom or corporate images that you use in your deployment you can just include them, so they will be downloaded and pre-cached in the installation stage.